### DISCUSSION TITLE
help with json save

### DISCUSSION THREAD

**Discussion by danelerr** [clientIO/joint#2498] [Q&A]
### Introduction

I am using the sequence diagram demo, but when I save the diagram in json and reload it some of the features are not working anymore.

I have created a button that when clicked does this

`
   document.getElementById('bc').addEventListener('click', () => {
        paper.freeze();
        graph.fromJSON(graph.toJSON());
        paper.unfreeze();
    });
`

and just after clicking, everything works, except for the connection of sd.Message between sd.Lifeline, I cannot add new sd.Message. 

the event that should do what I want is this one

`
    graph.on('add', (link) => {
        if (!link.isLink()) return;
        const type = link.get('type');
        switch (type) {
            case 'sd.Lifeline': {
                const tools = new dia.ToolsView({
                    layer: null,
                    tools: [
                        new linkTools.HoverConnect({
                            scale: toolsScale
                        }),
                    ]
                });
                link.findView(paper).addTools(tools);
                break;
            }
        }
    });
`

### Steps to reproduce

just use the demo code from the sequence diagrams, and try to save it and then load it.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
`graph.fromJSON()` does not trigger `add` events for individual cells. It only triggers a single `reset` event on graph.

But you can do this instead:
```js
graph.fromJSON(graph.toJSON());
graph.getLinks().forEach(link => {
  const type = link.get('type');
  switch (type) {
    case 'sd.Lifeline': {
      const tools = new dia.ToolsView({
        layer: null,
        tools: [ new linkTools.HoverConnect({ scale: toolsScale }), ]
      });
      link.findView(paper).addTools(tools);
      break;
     }
   }
});
```
